src/test/scala/CoreServerTests.scala

```diff
@@ -0,0 +1,340 @@
+import akka.actor.{Actor, ActorSystem, Props}
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import org.scalatest.{Matchers, WordSpec}
+import viper.server.ViperBackendConfigs.SiliconConfig
+import viper.server.{AstGenerator, VerificationJobHandler, ViperConfig, ViperCoreServer}
+import viper.silver.logger.{SilentLogger, ViperStdOutLogger}
+import viper.silver.reporter._
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+
+
+class CoreServerTest extends WordSpec with Matchers with ScalatestRouteTest {
+  import scala.language.postfixOps
+
+  implicit var actor_system: ActorSystem = ActorSystem("Test")
+  val client_actor = actor_system.actorOf(ClientActor.props())
+
+  object ClientActor {
+    case object Terminate
+    def props(): Props = Props(new  ClientActor())
+  }
+
+  class ClientActor() extends Actor{
+
+    override def receive: PartialFunction[Any, Unit] = {
+      case _: OverallSuccessMessage =>
```

Comment:
For the streaming test case, please add a list of expected outputs that the client actor gets as inout argument. Before the client actor terminates itself, it should assert that actual outcomes match the expected outcomes. In this scenario, I suggest using an input with 3 methods, two of which verify and one fails, or something like that.